packages/babel-types/src/clone/cloneNode.js

```diff
@@ -0,0 +1,54 @@
+import { NODE_FIELDS } from "../definitions";
+
+const has = Function.call.bind(Object.prototype.hasOwnProperty);
+const isNode = obj => obj && has(NODE_FIELDS, obj.type);
+
+function cloneIfNodeOrArray(obj) {
+  if (isNode(obj)) return cloneNode(obj);
+  if (Array.isArray(obj)) return obj.map(cloneIfNodeOrArray);
```

Comment:
I don't think there are cases were we'd ever have arrays of arrays, so we can probably have this make `cloneNode` instead of `cloneIfNodeOrArray`?